### PR TITLE
#cu-20465559 #ibc #cosmwasm module integration into #dali runtime

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
made coswasm to impl latest ibc traits and started to route pallet calls to cosmwasm on dali

### CODE REVIEW

File: code/parachain/frame/cosmwasm/src/ibc.rs [hunk 1]
```diff
@@ -159,11 +159,17 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
-#[derive(Default, RuntimeDebug, Eq, PartialEq, Clone)]
+#[derive(RuntimeDebug, Eq, PartialEq, Clone)]
 pub struct Router<T: Config> {
 	_marker: PhantomData<T>,
 }
 
+impl<T: Config> Default for Router<T> {
```
Comment (blasrodri) [inline code/parachain/frame/cosmwasm/src/ibc.rs:167]: why not deriving Default?

**Review by blasrodri — APPROVED** [review]

File: code/parachain/frame/cosmwasm/src/ibc.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/parachain/frame/cosmwasm/src/ibc.rs:167]: as I recall runtme macro does not works with attributes.

**Review by obsessed-cake — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@andyjsbell please take a look

**Review by andyjsbell — APPROVED** [review]
thanks @dzmitry-lahoda

**Pull request merged**